src/main/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpoints.scala

```diff
@@ -4,14 +4,21 @@ import cats.effect.Effect
 import cats.implicits._
 import io.circe.generic.auto._
 import io.circe.syntax._
-import io.github.pauljamescleary.petstore.domain.UserAlreadyExistsError
-import io.github.pauljamescleary.petstore.domain.users.{User, UserService}
 import org.http4s.circe._
-import org.http4s.{EntityDecoder, HttpService}
 import org.http4s.dsl.Http4sDsl
+import org.http4s.{EntityDecoder, HttpService, QueryParamDecoder}
+
+import scala.language.higherKinds
+import io.github.pauljamescleary.petstore.domain.{UserAlreadyExistsError, UserNotFoundError}
+import io.github.pauljamescleary.petstore.domain.users.{User, UserService}
 
 class UserEndpoints[F[_]: Effect] extends Http4sDsl[F] {
 
+  import QueryParamDecoder._
+
+  object PageSizeMatcher extends QueryParamDecoderMatcher[Int]("pageSize")
```

Comment:
we are using the same decoders pageSize and offset for pets.  Perhaps pull these into a separate module and import in both?

Reply:
By separate module you mean trait i.e. named 'Pagination' ?

Reply:
An object Pagination would be sufficient, and we could import the matchers.